static/js/buddy_data.js

```diff
@@ -327,14 +327,12 @@ function filter_user_ids(user_filter_text, user_ids) {
             return false;
         }
 
-        if (muting.is_user_muted(user_id)) {
-            // Muted users are hidden from the right sidebar entirely.
-            return false;
-        }
-
         return true;
     });
 
+    // Muted users are hidden from the right sidebar entirely.
+    user_ids = muting.filter_muted_user_ids(user_ids);
```

Comment:
I think we don't want to do this particular refactor, basically because `user_ids` is a huge list that we rerender frequently, and it is probably worth avoiding unnecessary copies of the list.